Notebook catalog documentation (#830)
smart-data-lake/smart-data-lake: docs/reference/notebookCatalog.md

---
id: notebookCatalog
title: Use in Notebooks
---

> SDLB can also be used in Notebooks. On one side SDLB can provide a high-quality catalog of data objects, and on the other hand it helps to easily prototype transformations in an interactive Notebook environment, and then transfer them to repository for production use.

Reviewer: ```suggestion
SDLB can also be used in Notebooks. On one side SDLB can provide a high-quality catalog of data objects, and on the other side it helps to easily prototype transformations in an interactive Notebook environment, and then transfer them to repository for production use.
```

                    <execution>
                        <id>compile-catalog</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>compile</goal>
                        </goals>
                        <configuration>
                            <sourceDir>./src/main/scala-generated</sourceDir>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</profile>
```

The LabCatalogGenerator creates source code for class `ActionCatalog` and `DataObjectCatalog` in package `io.smartdatalake.generated`. These names can be customized by command line parameters. The directory for the generated source code has to be provided as `srcDirectory` parameter.

Once generated, the fat-jar (created with maven profile `fat-jar`) and SDLB configuration files have to be uploaded into a Databricks Volume. Then create a Databricks Cluster with the fat-jar as Library and Environment Variable `JNAME=zulu17-ca-amd64` to use Java 17.

> Make your SDLB data pipeline did run at least once, so that tables are existing. A simple Databricks Job configuration can look as follows:

Reviewer: ```suggestion
Make sure your SDLB data pipeline runs at least once to ensure that tables exist. A simple Databricks Job configuration can look as follows:
```
